src/System.CommandLine/ParseResult.cs

```diff
@@ -72,7 +73,8 @@ private void AddImplicitOptionsAndCheckForErrors()
                         }
                     }
 
-                    if (command.Command.Argument.HasDefaultValue &&
+                    if (command.Command.Argument != null &&
```

Comment:
I believe we're disallowing or should disallow a null `Argument` property, because the distinction between that and `Argument.None` (which is the default) is unclear.

Reply:
I'm fine with that and removed this check and tests pass. 

Do we need an issue to ensure that we never get this far, because it's a bad place for a user to crash? Or do you think that won't recur